src/Mod/PartDesign/fcsprocket/sprocket.py

```diff
@@ -0,0 +1,144 @@
+# (c) 2020 Adam Spontarelli <adam@vector-space.org>
+#
+#   This program is free software; you can redistribute it and/or modify
+#   it under the terms of the GNU Lesser General Public License (LGPL)
+#   as published by the Free Software Foundation; either version 2 of
+#   the License, or (at your option) any later version.
+#   for detail see the LICENCE text file.
+#
+#   FCGear is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU Library General Public License for more details.
+#
+#   You should have received a copy of the GNU Library General Public
+#   License along with FCGear; if not, write to the Free Software
+#   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307
+
+from math import cos, sin, tan, sqrt, radians, acos, atan, asin, degrees
+import math
+
+import sys
+if sys.version_info.major >= 3:
+    xrange = range
```

Comment:
There's really no point in doing this. This distinction only truly makes sense if you are iterating over thousands or millions of objects. Besides, Python 2 support if basically over, so now we should just use `range` without issues everywhere.